src/frontend/src/common.tsx

```diff
@@ -560,8 +595,42 @@ export const UserLink = ({
     );
 };
 
-export const userList = (ids: UserId[] = []) =>
-    commaSeparated(ids.map((id) => <UserLink key={id} id={id} />));
+// We use u64::MAX to denote anonymous users. We should filter them out before we try to load user data.
+const removeNAs = (ids: UserId[]) =>
+    ids.filter((id) => id < Number.MAX_SAFE_INTEGER);
+
+export const UserList = ({
+    ids = [],
+    loadedNames = {},
+    profile,
+}: {
+    ids: UserId[];
+    loadedNames?: UserData;
+    profile?: boolean;
+}) => {
+    const [names, setNames] = React.useState<UserData>(loadedNames);
+
+    const loadNames = async () =>
+        setNames(
+            (await window.api.query<UserData>("users_data", removeNAs(ids))) ||
+                {},
+        );
+
+    React.useEffect(() => {
+        if (removeNAs(ids) && Object.entries(loadedNames).length == 0)
+            loadNames();
+    }, []);
+
+    return Object.keys(names).length == 0 && removeNAs(ids).length > 0 ? (
```

Comment:
Nit: this check is inconsistent with the similar check in line 620.

Reply:
Thanks, yes, this could be improved to not load an empty list.